app/javascript/mastodon/features/home_timeline/index.jsx

```diff
@@ -28,12 +31,36 @@ const messages = defineMessages({
   hide_announcements: { id: 'home.hide_announcements', defaultMessage: 'Hide announcements' },
 });
 
+const getHomeFeedSpeed = createSelector([
+  state => state.getIn(['timelines', 'home', 'items'], ImmutableList()),
+  state => state.get('statuses'),
+], (statusIds, statusMap) => {
+  const statuses = statusIds.take(20).map(id => statusMap.get(id));
+  const uniqueAccountIds = (new Set(statuses.map(status => status.get('account')).toArray())).size;
+  const oldest = new Date(statuses.getIn([statuses.size - 1, 'created_at'], 0));
+  const newest = new Date(statuses.getIn([0, 'created_at'], 0));
+  const averageGap = (newest - oldest) / (1000 * (statuses.size + 1)); // Average gap between posts on first page in seconds
+
+  return {
+    unique: uniqueAccountIds,
+    gap: averageGap,
+    last: newest,
+  };
+});
+
+const homeTooSlow = createSelector(getHomeFeedSpeed, speed =>
+  speed.unique < 7 // If there are fewer than 7 different accounts visible
+  || speed.gap > (20 * 60) // If the average gap between posts is more than 20 minutes
+  || (Date.now() - speed.last) > (1000 * 60 * 60) // If the most recent post is from over an hour ago
+);
```

Comment:
Looking at these metrics…
- I believe the 20 last posts being made by at least 7 different people is very likely to trigger even for people with a very active feed
- this will likely trigger in the morning for most people who do not follow people from many timezones
- more generally, the whole “less than a post every 20 minutes is too slow” approach is a really worrying mindset, though I understand the point of the feature, and the wording of the banner is not bad

Reply:
I tried basing this on my feed, though of course it depends on time of day. I don't follow that many people (413).

Reply:
Changed to 5 unique accounts and 30 min gap.

Reply:
I think the 5 unique accounts thing needs to go. Otherwise, “It's looking pretty quiet right now” will pop up when a limited set of followed people are having a lively conversation, which does not make a lot of sense.